x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/frequentitemsets/CountingItemSetTraverser.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.aggs.frequentitemsets;
+
+import org.apache.lucene.util.BitSet;
+import org.apache.lucene.util.FixedBitSet;
+import org.apache.lucene.util.LongsRef;
+import org.elasticsearch.core.Releasable;
+import org.elasticsearch.core.Releasables;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+/**
+ * Item set traverser to find the next interesting item set.
+ *
+ * The traverser returns items that we haven't visited in this combination yet.
+ *
+ *  basic algorithm
+ *
+ *  - expand the set with the item reported by the traverser
+ *  - re-calculate the count of transactions that contain the given item set
```

Comment:
Bit of a nit, but Javadoc is written in HTML, not markdown.  This should be <ul><li></li></ul> format.  Likewise, paragraph breaks should have <p> tags.

Applies to many javadoc blocks in this PR.

Reply:
Intellij and Eclipse both render javadoc on hover, which is how a lot of folks read it.  It shouldn't be necessary to click through to the class to get a correct representation.  You can use `<pre>` tags to get the graphs looking the way you want them to when rendered, I think.  Also, most of the graphs look like they're inline, which shouldn't be javadoc anyway.